### DISCUSSION TITLE
LXQT parent source code has empty folders

### DISCUSSION THREAD

**Discussion by thiyashoksanju** [lxqt/lxqt#2252] [Q&A] [answered]
I tried to download entire source code of lxqt 0.17.0 from [here](https://github.com/lxqt/lxqt/tree/0.17.0).
But the downloaded zip file has only empty directories of all lxqt sub components. 
I have to step into every sub projects individually and download their code to get the source.
Could you please solve this single click download for above mentioned link

**Comment by stefonarch** [comment]
Afaik this repo is meant to build only latest git of LXQt (using `git clone  ` and the commands in the README) and is  not suitable for older versions.

**Reply by thiyashoksanju** [reply]
We are working on lxqt o.17.0 to do some customization. For that purpose we need the source code of specified version.

**Comment by kalgnana** [comment]
Hi Lxqt Team,

As we do single compilation step to compile entire Linux kernel sub-system,  do we have any similar way to download and compile whole 46 Lxqt repository by single command ?

**Reply by stefonarch** [reply]
I've used time ago [a script to compile all components](https://github.com/lxqt/lxqt.git) - you could adapt it (at least for the components which has version numbering 0.17, using
```
-...
wget https://github.com/lxqt/$i/archive/refs/tags/0.17.0.zip
unzip $i-0.17.0.zip
cd $i-0.17.0
...
```
But I see that any  `lxqt-<version>.zip|.tar.gz` downloaded from https://github.com/lxqt/lxqt/releases/ is empty too. @tsujan ?

**Comment by tsujan** [comment, marked as answer]
> I tried to download entire source code of lxqt 0.17.0 from [here](https://github.com/lxqt/lxqt/tree/0.17.0).

https://github.com/lxqt/lxqt is for a one-step compilation of *git sources* (see https://github.com/lxqt/lxqt/wiki/Building-from-source#scripts-to-build-in-one-step); you can't get the latest release with it, let alone an outdated version like 0.17. We don't add those `zip` and `tar.gz` packages to https://github.com/lxqt/lxqt/releases; GitHub does.

Please also note that old versions of LXQt are considered dead. They aren't supported in any way. But if, for whatever reason, you want to compile and install an outdated version (which is not recommended), read https://github.com/lxqt/lxqt/wiki/Building-from-source#compiling-all-components-individually.
